sdk/core/src/main/java/com/klaviyo/core/KLog.kt

```diff
@@ -0,0 +1,116 @@
+package com.klaviyo.core
+
+import android.os.Build
+import com.klaviyo.core.config.Config
+import com.klaviyo.core.config.Log
+import com.klaviyo.core.config.Log.Level
+import java.util.regex.Pattern
+
+open class KLog : Log {
+
+    private var _logLevel: Level? = null
+    override var logLevel: Level
+        get() {
+            if (_logLevel == null && Registry.isRegistered<Config>()) {
+                _logLevel = Registry.config.getManifestInt(LOG_LEVEL, defaultLogLevel.ordinal).let {
+                    Level.entries.getOrNull(it) ?: defaultLogLevel.also { default ->
+                        default.log(
+                            "KlaviyoLog",
+                            "Invalid log level $it detected in manifest, defaulting to $default."
+                        )
+                    }
+                }
+            }
+
+            return _logLevel ?: defaultLogLevel
+        }
+        set(value) {
+            _logLevel = value
+        }
+
+    override fun wtf(message: String, ex: Throwable?) = log(message, Level.Assert, ex)
+
+    override fun error(message: String, ex: Throwable?) = log(message, Level.Error, ex)
+
+    override fun warning(message: String, ex: Throwable?) = log(message, Level.Warning, ex)
+
+    override fun info(message: String, ex: Throwable?) = log(message, Level.Info, ex)
+
+    override fun debug(message: String, ex: Throwable?) = log(message, Level.Debug, ex)
+
+    override fun verbose(message: String, ex: Throwable?) = log(message, Level.Verbose, ex)
+
+    fun log(msg: String, level: Level, ex: Throwable? = null) {
+        if (logLevel == Level.None) return
+
+        if (level.ordinal >= logLevel.ordinal) {
+            level.log(makeTag(), msg, ex)
+        }
+    }
+
+    private companion object {
+        /**
+         * Log level key for manifest, specifying the minimum log level to output:
+         * 0 = disable logging entirely
+         * 1 = Assert only
+         * 2 = Error and above
+         * 3 = Warning and above
+         * 4 = Info and above
+         * 5 = Debug and above
+         * 6 = Verbose and above
```

Comment:
these seem inconsistent from the readme

Reply:
dangit how many places did I write this...

Reply:
@kennyklaviyo fixed , thx